apis/buzzer/src/lib.rs

```diff
@@ -0,0 +1,173 @@
+//! Implementation started by : https://github.com/teodorobert
+//! Continued and modified by : https://github.com/SheepSeb
+#![no_std]
+
+use core::cell::Cell;
+
+use libtock_platform::{
+    share, subscribe::OneId, DefaultConfig, ErrorCode, Subscribe, Syscalls, Upcall,
+};
+pub struct Buzzer<S: Syscalls>(S);
+
+impl<S: Syscalls> Buzzer<S> {
+    /// Returns Ok() if the driver was present.This does not necessarily mean
+    /// that the driver is working.
+    pub fn exists() -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, EXISTS, 0, 0).to_result()
+    }
+
+    /// Initiate a tone
+    pub fn tone(freq: u32, duration: u32) -> Result<(), ErrorCode> {
```

Comment:
```suggestion
    pub fn tone(freq: u32, duration: Duration) -> Result<(), ErrorCode> {
```

Reply:
Use the [`core::time::Duration`](https://doc.rust-lang.org/core/time/struct.Duration.html) data type.